frontend/src/pages/pipelines/global/PipelineCoreApplicationPage.tsx

```diff
@@ -4,7 +4,7 @@ import NoPipelineServer from '~/concepts/pipelines/NoPipelineServer';
 import PipelineCoreProjectSelector from '~/pages/pipelines/global/PipelineCoreProjectSelector';
 import { PipelineServerTimedOut, usePipelinesAPI } from '~/concepts/pipelines/context';
 
-type PipelineCoreApplicationPageProps = {
+export type PipelineCoreApplicationPageProps = {
```

Comment:
You never have to export props -- grab it from `React.ComponentProps<typeof TheComponent>`

Reply:
Ok, so that's a rule we want to establish? Never export props?

Reply:
Updated.

Reply:
Deriving props from the component is far better than exporting a type out of the same file. It's just cleaner and less maintenance if things had to change or shift in anyway. You're going straight to the source.